app/services/notifications/reactions/send.rb

```diff
@@ -41,7 +41,8 @@ def call
         end
 
         if aggregated_reaction_siblings.size.zero?
-          notification = Notification.where(notification_params).delete_all
+          Notification.where(notification_params).delete_all
+          OpenStruct.new(action: :deleted)
```

Comment:
I know `OpenStruct.new(action: :deleted).notification_id` will result in `nil` but maybe it's better to say it explicitly, because for a moment I was wondering why the signature of this struct was different from the other pone below.

Keep in mind that OpenStruct is much, much slower than `Struct` and than a normal hash: https://stackoverflow.com/a/4459132/4186181

Reply:
These are valid points, I am not 100% happy with the current solution too. Maybe it's worth refactoring in another pr (since this one is merged already)